src/App/Utilities/AccountManagement/AccountsManager.cs

```diff
@@ -0,0 +1,227 @@
+                    await SwitchedAccountAsync();
+                    break;
+            }
+        }
+
+        private void Locked(Tuple<string, bool> extras)
+        {
+            var userId = extras?.Item1;
+            var userInitiated = extras?.Item2 ?? false;
+            Device.BeginInvokeOnMainThread(async () => await LockedAsync(userId, userInitiated));
+        }
+
+        private async Task LockedAsync(string userId, bool userInitiated)
+        {
+            if (!await _stateService.IsActiveAccountAsync(userId))
+            {
+                _platformUtilsService.ShowToast("info", null, AppResources.AccountLockedSuccessfully);
+                return;
+            }
+
+            var autoPromptBiometric = !userInitiated;
+            if (autoPromptBiometric && Device.RuntimePlatform == Device.iOS)
+            {
+                var vaultTimeout = await _stateService.GetVaultTimeoutAsync();
+                if (vaultTimeout == 0)
+                {
+                    autoPromptBiometric = false;
+                }
+            }
+
+            await _accountsManagerHost.SetPreviousPageInfoAsync();
+            
+            Device.BeginInvokeOnMainThread(() => _accountsManagerHost.Navigate(NavigationTarget.Lock, new LockNavigationParams(autoPromptBiometric)));
+        }
+
+        private void AddAccount()
+        {
+            Device.BeginInvokeOnMainThread(() =>
+            {
+                Options.HideAccountSwitcher = false;
+                _accountsManagerHost.Navigate(NavigationTarget.HomeLogin);
+            });
+        }
+
+        private void LogOut(Tuple<string, bool, bool> extras)
+        {
+            var userId = extras?.Item1;
+            var userInitiated = extras?.Item2 ?? true;
+            var expired = extras?.Item3 ?? false;
+            Device.BeginInvokeOnMainThread(async () => await LogOutAsync(userId, userInitiated, expired));
+        }
+
+        private async Task LogOutAsync(string userId, bool userInitiated, bool expired)
+        {
+            await AppHelpers.LogOutAsync(userId, userInitiated);
+            await NavigateOnAccountChangeAsync();
+            _authService.LogOut(() =>
+            {
+                if (expired)
+                {
+                    _platformUtilsService.ShowToast("warning", null, AppResources.LoginExpired);
+                }
+            });
+        }
+
+        private async Task SwitchedAccountAsync()
+        {
+            await AppHelpers.OnAccountSwitchAsync();
+            Device.BeginInvokeOnMainThread(async () =>
+            {
+                if (await _vaultTimeoutService.ShouldTimeoutAsync())
+                {
+                    await _vaultTimeoutService.ExecuteTimeoutActionAsync();
+                }
+                else
+                {
+                    await NavigateOnAccountChangeAsync();
+                }
+                await Task.Delay(50);
```

Comment:
I figured it would be something of sorts, my fear is that the 50ms may be enough for some cases but not for all.

Reply:
Just in case this was taken from  an already working place and so far no issues were raised regarding this. For now I'd leave it as is and if somebody complain we can change it